File: C_Programming/Competitive/Assignment41_5.c
/*
    Write a recursive program which accept number from user and return its
    product of digits.
    Input : 523
    Output : 30
*/

#include<stdio.h>

int Mult(int iNo)
{
    static int iMult = 1 ;
    int iDigit = 0 ;
    if(iNo != 0 )
    {
        iDigit = iNo % 10 ;

        iMult = iMult * iDigit ;
        iNo = iNo / 10 ;

        Mult(iNo);
    }
    return iMult ;
}

int main()
{
    int iValue = 0 ;
    int iRet = 0 ;

    printf("Enter number : ");
    scanf("%d",&iValue);

    iRet = Mult(iValue);

    printf("Result is : %d\n",iRet);

    return 0 ;
}

File: Java_Programming/Competitive/Assignment41_5.java

import java.util.Scanner;

class Assignment41_5
{
    static int iProduct = 1 ; 

//--------------------------------------------------------------------//
// Function Name : Mult
// Description   : Calculates the product of digits recursively.
// Input         : Integer number
// Output        : Product of digits
// Author        : Pranav Avinash Narkhede
// Date          : 24-07-2026
//--------------------------------------------------------------------//
    public static int Mult(int iNo)
    {
        int iDigit = 0 ;

        if(iNo != 0)
        {
            iDigit = iNo % 10 ;

            iProduct = iDigit * iProduct;

            iNo = iNo / 10 ;

            Mult(iNo);
        }

        return iProduct;
        
    }

//--------------------------------------------------------------------//
// Application to calculate the product of digits using recursion.
//--------------------------------------------------------------------//
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0 ;
        int iRet = 0 ;

        System.out.println("Enter the number : ");
        iValue = sobj.nextInt();

        iRet = Mult(iValue);

        System.out.println("Product of digits : "+iRet);
    }
}